R/Logger.R

```diff
@@ -263,3 +426,48 @@ Logger <- R6::R6Class( #nolint: object_name_linter, cyclocomp_linter
     }
   )
 )
+
+
+#' LoggerNull: The no-logging Logger
+#'
+#' @description
+#'   The `LoggerNull` class overwrites the functions of the `Logger` so no logging is produced.
+#'   Errors and warnings are still produced.
+#' @return
+#'   A new instance of the `LoggerNull` [R6][R6::R6Class] class.
+#' @examples
+#'   logger <- LoggerNull$new()
+#'
+#'   logger$log_info("This message will not print!")
+#'   logger$log_to_db(message = "This message will no be written in DB!")
+#'   try(logger$log_warn("This is a warning!"))
+#'   try(logger$log_error("This is an error!"))
+#' @export
+#' @importFrom R6 R6Class
+LoggerNull <- R6::R6Class(                                                                                              # nolint: object_name_linter
+  classname  = "LoggerNull",
+  inherit = Logger,
+  public = list(
+
+    #' @description
+    #'   Create a new `LoggerNull` object
+    #' @param ... Captures arguments given, but does nothing
+    initialize = function(...) {
+      super$initialize(output_to_console = FALSE, warn = FALSE) # Disable all console logging
```

Comment:
```suggestion
super$initialize(output_to_console = FALSE, warn = FALSE, log_path = NULL) # Disable all console logging

```
log_path = NULL is needed in the case it is given in .Rprofile